src/plugins/highlight/findHighlightMarkers.ts

```diff
@@ -21,24 +21,41 @@ import { HighlightMarker, HighlightStartMarker } from './types'
 export function findHighlightMarkers(doc: PMNode) {
   const markers: HighlightMarker[] = []
   let current: HighlightStartMarker | undefined
+
+  function isCommentNode(node: PMNode) {
+    return node.type.name === 'comment'
+  }
+
   doc.descendants((node, pos) => {
+    const commentIds: string[] = []
+
+    doc.descendants((node) => {
```

Comment:
use `comment_list` so it will be like this:
```
if(node.type.name === schema.nodes.comment_list){
  node.forEach(({attrs}) => {
     commentIds.push(attrs.id)
  })
  return false
}
```
and when we return false in `descendants` it's going to stop the loop and move that out of `descendants` to avoid O(n²) time complexity

Reply:
@asouqi 
Thank you for highlighting the complexity, it really needs an enhancement. I made additional modifications to avoid using extra loops. Kindly review and let me know your thoughts. thanks